x/committee/keeper/grpc_query.go

```diff
@@ -0,0 +1,183 @@
+}
+
+// Committee implements the Query/Committee gRPC method.
+func (k QueryHandler) Committee(c context.Context, req *types.QueryCommitteeRequest) (*types.QueryCommitteeResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(c)
+	committee, found := k.GetCommittee(ctx, req.CommitteeID)
+	if !found {
+		return nil, status.Errorf(codes.NotFound, "could not find committee for id: %v", req.CommitteeID)
+	}
+	committeeAny, err := types.PackCommittee(committee)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "could not pack committees: %v", err)
+	}
+	return &types.QueryCommitteeResponse{Committee: committeeAny}, nil
+}
+
+// Proposals implements the Query/Proposals gRPC method
+func (k QueryHandler) Proposals(c context.Context, req *types.QueryProposalsRequest) (*types.QueryProposalsResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(c)
+	proposals := k.GetProposalsByCommittee(ctx, req.CommitteeID)
+	proposalsResp := types.QueryProposalsResponse{
+		Proposals: make([]types.QueryProposalResponse, len(proposals)),
+	}
+	for i, proposal := range proposals {
+		proposalsResp.Proposals[i] = k.proposalResponseFromProposal(proposal)
+	}
+
+	return &proposalsResp, nil
+}
+
+// Proposal implements the Query/Proposal gRPC method
+func (k QueryHandler) Proposal(c context.Context, req *types.QueryProposalRequest) (*types.QueryProposalResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(c)
+	proposal, found := k.GetProposal(ctx, req.ProposalID)
+	if !found {
+		return nil, status.Errorf(codes.NotFound, "cannot find proposal: %v", req.ProposalID)
+	}
+	proposalResp := k.proposalResponseFromProposal(proposal)
+	return &proposalResp, nil
+}
+
+// NextProposalID implements the Query/NextProposalID gRPC method
+func (k QueryHandler) NextProposalID(c context.Context, req *types.QueryNextProposalIDRequest) (*types.QueryNextProposalIDResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(c)
+	proposalID, err := k.GetNextProposalID(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "cannot find next proposal id: %v", err)
+	}
+
+	return &types.QueryNextProposalIDResponse{NextProposalID: proposalID}, nil
+}
+
+// Votes implements the Query/Votes gRPC method
+func (k QueryHandler) Votes(c context.Context, req *types.QueryVotesRequest) (*types.QueryVotesResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(c)
+	votes := k.GetVotesByProposal(ctx, req.ProposalID)
+	votesResp := types.QueryVotesResponse{
+		Votes: make([]types.QueryVoteResponse, len(votes)),
+	}
```

Comment:
For the token committee the number of votes could get pretty large. So it's probably worth implementing pagination on this new votes endpoint.

I can't imagine a scenario where we'd have enough committees to be worth paginating. There's quite a few proposals, but this endpoint only returns the active ones, which there probably won't be many of. So only paginating votes seems ok.

Most of the cosmos modules use a Paginate function in their grpc servers.

Reply:
agreed, I can add this in